packages/client/src/runtime.ts

```diff
@@ -1,65 +1,92 @@
-import { Observable, combineLatest, filter, map } from "rxjs"
 import {
   ChainHead$,
   RuntimeContext,
   getObservableClient,
 } from "@polkadot-api/observable-client"
+import {
+  Observable,
+  ReplaySubject,
+  combineLatest,
+  connectable,
+  filter,
+  firstValueFrom,
+  map,
+} from "rxjs"
+import { DescriptorsValue } from "./descriptors"
+
+export enum OpType {
+  Storage = 0,
+  Tx = 1,
+  Event = 2,
+  Error = 3,
+  Const = 4,
+}
 
 export class Runtime {
-  protected _ctx: unknown
-  protected _checksums: string[]
+  private constructor(
+    private _ctx: RuntimeContext,
+    private _checksums: string[],
+    private _descriptors: DescriptorsValue,
+  ) {}
 
-  private constructor(ctx: RuntimeContext, checksums: string[]) {
-    this._ctx = ctx
-    this._checksums = checksums
+  /**
+   * Internal implementation detail. Do not use
+   */
+  static _create(
+    ctx: RuntimeContext,
+    checksums: string[],
+    descriptors: DescriptorsValue,
+  ) {
+    return new Runtime(ctx, checksums, descriptors)
+  }
+
+  /**
+   * Internal implementation detail. Do not use
+   */
+  _getCtx() {
+    return this._ctx
+  }
+
+  /**
+   * Internal implementation detail. Do not use
+   */
+  _getPalletChecksum(opType: OpType, pallet: string, name: string) {
+    return this._checksums[this._descriptors.pallets[pallet][opType][name]]
+  }
+
+  /**
+   * Internal implementation detail. Do not use
+   */
+  _getApiChecksum(name: string, method: string) {
+    return this._checksums[this._descriptors.apis[name][method]]
```

Comment:
Maybe `@private` could be used. https://jsdoc.app/tags-private

Reply:
Nice, I didn't know. I will add that in 👍